### ISSUE TITLE
Amazon Redshift support

### ISSUE THREAD

**Issue by izevaka** [fluentmigrator/fluentmigrator#604]
Amazon Redshift _almost_ works with fluentmigrator. Redshift is Amazon's data warehouse product whose SQL dialect is loosely based on Postgres 8.0.2 - http://docs.aws.amazon.com/redshift/latest/dg/c_redshift-and-postgres-sql.html. 

There are two issues that prevent migrations from working out of the box, both to do with `VersionInfo` table.
### Casing of version table and its fields

Redshift names are folded to lower case (http://docs.aws.amazon.com/redshift/latest/dg/r_names.html).

> Standard and delimited identifiers are case-insensitive and are folded to lower case.

This means that mixed case identifiers created with `Create...` are converted to lower case. Coupled with Postgres' version of [TableExists](https://github.com/schambers/fluentmigrator/blob/master/src/FluentMigrator.Runner/Processors/Postgres/PostgresProcessor.cs#L44), the following code returns false, resulting in version table being recreated every time.

```
Create.Table("VersionInfo");
var tableExists = Processor.TableExists("schema", "VersionInfo");
//On Redshift tableExists is false
```

There is a workaround for this and that is to use a custom implementation of `IVersionTableMetaData` with all lower case identifiers.
### Unique index on VersionInfo.Version

This one is a deal breaker (at least I haven't worked out how to work around it in just the migration project). `CREATE INDEX` is not supported in Redshift - http://docs.aws.amazon.com/redshift/latest/dg/c_unsupported-postgresql-features.html. Migration runner fails on trying to create index and we are stuck.

There are two options that I can see that would enable Redshift support. Both would remove the unique index in the version table for redshift databases (which btw I don't really understand, even after rereading #219). 
### Option 1 - Redshift provider

Don't know how feasible this is. This is pretty much going to be a minimal copy of postgres one, with some changes to do case insensitive lookups and noop index and constraint operations. This seems like the best option as it isolates all the DB specific issues in the DB specialisation layer. 
### Option 2 - IVersionTableMetaData extension

This would be a variant of the following change - https://github.com/izevaka/fluentmigrator/commit/a962f8dcf68f5b946c6f75e180a51d3af703d04f, which demonstrates minimal amount of work required to support Redshift. This is a minimal core change, but pushes the redshift ugliness to the migration project. 

Im happy to do the work for either, just wanting to sanity check the approach.

**Comment by tommarien** [comment]
@izevaka Hi

It seems to me that option 2 is just a hack, whereas option 1 is the solution.

I can imagine more issues than just the version migration index issue. I' am guessing that a full fledged provider with quoter would be the most satifying in the end.

Kind regards

**Comment by anton89** [comment]
Sorry this a bit of out of context, but why postgre provider using quoter as a default?

In the past I just almost work with postgreSQL so I get to look around postgre provider, anyway a lot of googling show that it never been even best practice to put quote, also if you turn off those quoter than this problem shouldn't happen right?

**Comment by izevaka** [comment]
OK cool, I'll get cranking on option 1. Is there a guide for contributing that deals with integration test for different DB backends? I tried running tests in the solution and it seemed like they all wanted to connect to localhost for all different database types.

@anton89 Not sure the quoter is the culprit here. The bit that screws with casing is [this](https://github.com/schambers/fluentmigrator/blob/master/src/FluentMigrator.Runner/Processors/Postgres/PostgresProcessor.cs#L44):

```csharp
 return Exists("select * from information_schema.tables where table_schema = '{0}' and table_name = '{1}'", FormatToSafeSchemaName(schemaName), FormatToSafeName(tableName));
```

It should be:

```
return Exists("select * from information_schema.tables where table_schema ilike '{0}' and table_name ilike '{1}'", FormatToSafeSchemaName(schemaName), FormatToSafeName(tableName));
```

**Comment by izevaka** [comment]
PR #605 has been updated - I think that this is ready for review. In the meantime I will work on integration tests and Redshift specific additions - sort columns and distribution style clauses.

**Comment by fubar-coder** [comment]
Fixed in release/2.0.0
